File: frontend/LicitatieFrontend/src/app/Models/forum-post/forum-post.ts
import { ForumComment } from "../forum-comment/forum-comment";

export class ForumPost{
    id!: number;
    userId!: number;
    userName?: string;
    title!: string;
    date!: string;
    description!: string;
    commentsCount?: number;
    commentList?: ForumComment[];

    constructor( forumPost?: Partial<ForumPost>){
        Object.assign(this, forumPost);
    }
}
export interface CreateReportDto {
  targetType: 'ForumPost' | 'ForumComment';
  targetId: number;
  reason: string;
  reporterId: number;
}

export const REPORT_REASONS: string[] = [
  'Spam',
  'Limbaj licențios',
  'Conținut ofensator / hărțuire',
  'Informații false',
  'Alt motiv',
];

File: BackEnd/Backend/Backend/Models/ForumPost.cs
using System;
namespace Backend.Models
{
	public class ForumPost
	{
		public int Id { get; set; }
		public int UserId { get; set; }
		public DateTime Date { get; set; }
		public string Title { get; set; }
		public string Description { get; set; }
        //public List<ForumComment> Comments { get; set;  } = new List<ForumComment>();
		// TODO: Add photos
    }
}
